sql/hive/src/test/scala/org/apache/spark/sql/sources/SimpleTextRelation.scala

```diff
@@ -90,6 +90,7 @@ class SimpleTextSource extends TextBasedFileFormat with DataSourceRegister {
         }.reduceOption(expressions.And).getOrElse(Literal(true))
         InterpretedPredicate.create(filterCondition, inputAttributes)
       }
+      predicate.initialize(0)
```

Comment:
The predicate here can only be `sources.GreaterThan`. It can't include non-deterministic expressions.

Reply:
thanks， modify it.